partiql-cli/src/main/kotlin/org/partiql/cli/pipeline/Pipeline.kt

```diff
@@ -38,7 +38,10 @@ internal class Pipeline private constructor(
         val result = listen(ctx.errorListener as AppPErrorListener) {
             parser.parse(source, ctx)
         }
-        return result.root
+        if (result.statements.size != 1) {
+            throw PipelineException("Expected exactly one statement, got: ${result.statements.size}")
+        }
+        return result.statements[0]
```

Comment:
Can you add a `parse` and `parseAll` so that customers do not need to always handle this. The parse can either call parseAll and do the assertion and throw IllegalArgument, or it can invoke the single-statement ANTLR rule resulting in an "expected EOF" type error. This would be _slightly_ more efficient.